ntoskrnl/lpc/connect.c

```diff
@@ -733,6 +747,15 @@ NtSecureConnectPort(OUT PHANDLE PortHandle,
     return Status;
 
 Failure:
+    if (KeReadStateSemaphore(&Thread->LpcReplySemaphore))
```

Comment:
As far as I can tell:
* The STATUS_PORT_CONNECTION_REFUSED case above (L726/740) should not wait on the semaphore
* The first failure case in L598/612 already does this wait
* The exception case in L631/645 should not wait (actually to me it looks like it shouldn't even `goto Failure`)

So this doesn't seem right. Can you double-check?